merkle/sparse_merkle_tree_test.go

```diff
@@ -287,46 +288,58 @@ func testSparseTreeFetches(t *testing.T, vec sparseTestVector) {
 	const rev = 100
 	w, tx := getSparseMerkleTreeWriterWithMockTX(rev)
 
-	e := make(map[string]string)
+	reads := make(map[string]string)
+	readMutex := sync.Mutex{}
 	leafNodeIDs := make([]storage.NodeID, 0)
-	// calculate the set of expected node reads.
-	for i, kv := range vec.kv {
-		keyHash := w.hasher.keyHasher([]byte(kv.k))
-		nodeID := storage.NewNodeIDFromHash(keyHash)
-		leafNodeIDs = append(leafNodeIDs, nodeID)
-		sibs := nodeID.Siblings()
-		t.Logf("hash %d: %s", i, nodeID.String())
-
-		// start with the set of siblings of all leaves:
-		for j := range sibs {
-			j := j
-			id := sibs[j].String()
-			pathNode := nodeID.String()[:len(id)]
-			if _, ok := e[pathNode]; ok {
-				// we're modifying both children of a node because two keys are
-				// intersecting, since both will be recalculated neither will be read
-				// from storage so we remove the previously set expectation for this
-				// node's sibling, and skip adding one for this node:
-				delete(e, pathNode)
-				continue
+
+	{
+		readMutex.Lock()
+
+		// calculate the set of expected node reads.
```

Comment:
This expectation setting is starting to look like a lot of work. In some ways it might be better if we had a fake storage but as previously discussed that's a maintenance overhead too.